### PR TITLE
Catch raises section in docstring

### PR DESCRIPTION
**Pull request by mishig25**
Unlike transformers docs, datasets docs has a lot of using `raises` section in docstring. Therefore, this PR implements catching raises sections in docstrings

### CODE REVIEW

File: src/doc_builder/convert_rst_to_mdx.py [hunk 1]
```diff
@@ -394,6 +396,34 @@ def parse_rst_docstring(docstring):
             lines.insert(idx, "</parameters>\n")
             idx += 1
 
+        # Raises section
```
Comment (sgugger) [inline src/doc_builder/convert_rst_to_mdx.py:399]: I can't comment on line 385 but the test there will need to include the `_re_raises` as well now, same for the test at line 440.

Might be worth grouping both the "Returns" and "Raises" in the same re pattern and then have subtest there? I'll let you decide.

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

File: src/doc_builder/convert_rst_to_mdx.py [hunk 2]
```diff
@@ -437,7 +440,11 @@ def parse_rst_docstring(docstring):
             # The line may contain the return type.
             return_type, lines[idx] = split_return_line(lines[idx])
             idx += 1
-            while idx < len(lines) and (is_empty_line(lines[idx]) or find_indent(lines[idx]) >= return_indent):
+            while idx < len(lines) and (
+                is_empty_line(lines[idx])
+                or find_indent(lines[idx]) >= return_indent
+                or _re_raises.search(lines[idx]) is not None
+            ):
```
Comment (mishig25) [inline src/doc_builder/convert_rst_to_mdx.py:447]: @sgugger let me know if the brackets look good 😊

Comment (sgugger) [reply src/doc_builder/convert_rst_to_mdx.py:447]: Ah, thanks for checking! It is not ;-)
Here is what we need:
```suggestion
            while idx < len(lines) and_re_raises.search(lines[idx]) is None and (
                is_empty_line(lines[idx])
                or find_indent(lines[idx]) >= return_indent
            ):
```

Comment (mishig25) [reply src/doc_builder/convert_rst_to_mdx.py]: thanks a lot!

**Pull request merged**